### PR TITLE
fix: Disable generic API key rule by default in loki.secretfilter

### PR DESCRIPTION
**Pull request by romain-gaillard**
#### PR Description
We have [recently merged](https://github.com/grafana/alloy/pull/1593#event-14380083833) a new component for alloy called `loki.secretfilter`. This component hasn't been released yet and will be in 1.5.0.

However, we have noticed that the inclusion of the `generic-api-key` rule by default creates too many false positive secrets, so we're changing the config so that it's disabled by default.

#### PR Checklist

- [ ] CHANGELOG.md updated -> Don't think it applies since the component hasn't been included in a release yet.
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

File: docs/sources/reference/components/loki/loki.secretfilter.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ The `types` argument is a map of secret types to look for. The values are used a
 
 The `redact_with` argument is a string that can use variables such as `$SECRET_NAME` (replaced with the matching secret type) and `$SECRET_HASH`(replaced with the sha1 hash of the secret).
 
-The `exclude_generic` argument is a boolean that excludes the generic API key rule in the Gitleaks configuration file if set to `true`.
+The `include_generic` argument is a boolean that includes the generic API key rule in the Gitleaks configuration file if set to `true`. It is disabled by default because it can generate false positives.
```
Comment (wildum) [inline docs/sources/reference/components/loki/loki.secretfilter.md:54]: should not remove it completely if it's not good?

Comment (romain-gaillard) [reply docs/sources/reference/components/loki/loki.secretfilter.md:54]: It can be really useful! But because it's a generic secret-detection regex, it's definitely not perfect and some logs will have a format that sometimes triggers it (especially when they contain keywords like `secret`).

My opinion is that it should be disabled by default and enabled explicitly when the user has confidence that their log format doesn't generate false positives (or they excluded them), so that it can start picking up real secrets that weren't caught by the more specialised rules.

So in short, it can definitely be useful but requires a bit more work from the user.

**Review by wildum — APPROVED** [review]

**Comment by wildum** [comment]
LGTM, waiting on @clayton-cornell's approval before merging

File: docs/sources/reference/components/loki/loki.secretfilter.md (hunk 1 above)
Comment (clayton-cornell) [inline docs/sources/reference/components/loki/loki.secretfilter.md:54]:
```suggestion
The `include_generic` argument is a boolean that includes the generic API key rule in the Gitleaks configuration file if set to `true`. It's disabled by default because it can generate false positives.
```
